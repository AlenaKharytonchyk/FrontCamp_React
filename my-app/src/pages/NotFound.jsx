import React from 'react';
import SearchForm from "../Components/searchForm/searchForm";
import TitleText from "../Components/titleText/title";
import SearchBy from "../Components/searchBy/search";
import NoResults from "../Components/notFound/notFound";
import Footer from "../Components/footer/footer";

function NotFound() {

  return (
    <main className='main-container'>
      <header>
        <div className='header'>
          <TitleText/>
          <SearchForm/>
          <SearchBy name_One = 'TITLE' name_Two = 'GENRE' title='SEARCH BY'/>
        </div>
      </header>
      <section className='search-by'>
        <SearchBy name_One = 'RELEASE DATE' name_Two = 'RATING' title='SORT BY'/>
      </section>
      <section className='cards-section'>
        <NoResults/>
      </section>
      <Footer/>
    </main>
  );
}

export default NotFound;
