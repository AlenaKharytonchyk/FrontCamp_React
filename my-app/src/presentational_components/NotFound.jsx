import React from 'react';
import SearchForm from './searchForm/searchForm';
import TitleText from './titleText/title';
import SearchBy from '../shared_components/searchBy/search';
import NoResults from './notFound/notFound';
import PageContainer from '../shared_components/headerFooter/headerFooter';
import background from '../media/collage_.png';

const NotFound = () => (
  <PageContainer>
    <main className="main-container">
      <header style={{ backgroundImage: `url(${background})` }}>
        <div className="header">
          <TitleText />
          <SearchForm />
          <SearchBy name_One="TITLE" name_Two="GENRE" title="SEARCH BY" />
        </div>
      </header>
      <section className="search-by">
        <SearchBy name_One="RELEASE DATE" name_Two="RATING" title="SORT BY" />
      </section>
      <section className="cards-section">
        <NoResults />
      </section>
    </main>
  </PageContainer>
);

export default NotFound;
