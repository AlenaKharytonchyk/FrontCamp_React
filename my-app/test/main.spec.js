import React from 'react';
import {shallow} from "enzyme";
import SearchForm from "../src/presentational_components/searchForm/searchForm";
import SearchResultBy from "../src/shared_components/searchResult_forMovie_page/searchResultBy";
import Footer from "../src/shared_components/footer/footer";
import CardContainer from "../src/presentational_components/cardContainer/cardContainer";
import MovieCard from "../src/presentational_components/movieCard/movieCard";
import MovieList from "../src/presentational_components/movieList/movieList";
import NotFound from "../src/presentational_components/NotFound";
import TitleText from "../src/presentational_components/titleText/title";
import PageContainer from "../src/shared_components/headerFooter/headerFooter";
import SearchBy from "../src/shared_components/searchBy/search";
import SubmitButton from "../src/shared_components/submitBtn/submitBtn";

it('should render an search form', () => {
  const wrapper = shallow(
    <SearchForm click = {()=>{}}>Hello Jest!</SearchForm>
  );
  expect(wrapper).toMatchSnapshot();
});

it('should render a title text', () => {
  const wrapper = shallow(
    <TitleText/>
  );
  expect(wrapper).toMatchSnapshot();
});

describe('When Search by is rendered', function () {
  const obj = {
    name_One: "Name One",
    name_Two: "Name Two",
    title: "title",
    onClick: () => {},
  };

  it('should select first name when it is passed as selected', function () {
    const wrapper = shallow(
      <SearchBy selected={obj.name_One} {...obj}/>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should select second name when it is passed as selected', function () {
    const wrapper = shallow(
      <SearchBy selected={obj.name_Two} {...obj}/>
    );
    expect(wrapper).toMatchSnapshot();
  });

});

it('should render a title text', () => {



});

describe('When submit button is rendered', function () {
  it('should match snapshot', () => {
    const wrapper = shallow(
      <SubmitButton name='MyName' className='Class-name'/>
    );
    expect(wrapper).toMatchSnapshot();
  });
});

describe('When footer is rendered', function () {
  it('should match snapshot', () => {
    const wrapper = shallow(
      <Footer>Hello Jest!</Footer>
    );
    expect(wrapper).toMatchSnapshot();
  });
});

describe('When SearchResultBy is rendered', function () {
  let wrapper;
  beforeAll(function() {
    wrapper = shallow(
      <SearchResultBy value="Action" />
    );
  });

  it('should render as expected', function () {
    expect(wrapper).toMatchSnapshot();
  });

  it('should display title as expected', function () {
    expect(wrapper.text()).toEqual('Films by Action Genre')
  });
});

it('should render an search by filter', () => {
  const wrapper = shallow(
    <SearchResultBy>Hello Jest!</SearchResultBy>
  );
  expect(wrapper).toMatchSnapshot();
});

it('should render card container with genres', () => {
  const obj = {
    poster_path: 'path',
    title: 'title',
    id: 'id',
    genres: ['genres'],
    release_date: '20-09-2020',
  };
  const wrapper = shallow(
    <CardContainer {...obj}>Hello Jest!</CardContainer>
  );
  expect(wrapper).toMatchSnapshot();
});

it('should render card container without genres', () => {
  const obj = {
    poster_path: 'path',
    title: 'title',
    id: 'id',
    genres: [],
    release_date: '20-09-2020',
  };
  const wrapper = shallow(
    <CardContainer {...obj}>Hello Jest!</CardContainer>
  );
  expect(wrapper).toMatchSnapshot();
});

it('should render movie card', () => {
  const obj = {
    poster_path: 'path',
    title: 'title',
    id: 'id',
    release_date: '20-09-2020',
    vote_average: 'vote_average',
    runtime: 'runtime',
    overview: 'overview'
  };

  const wrapper = shallow(
    <MovieCard {...obj}>Hello Jest!</MovieCard>
  );
  expect(wrapper).toMatchSnapshot();
});

