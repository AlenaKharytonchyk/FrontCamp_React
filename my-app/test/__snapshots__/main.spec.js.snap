// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When Search by is rendered should select first name when it is passed as selected 1`] = `ShallowWrapper {}`;

exports[`When Search by is rendered should select second name when it is passed as selected 1`] = `ShallowWrapper {}`;

exports[`When SearchResultBy is rendered should render as expected 1`] = `ShallowWrapper {}`;

exports[`When footer is rendered should match snapshot 1`] = `ShallowWrapper {}`;

exports[`When submit button is rendered should match snapshot 1`] = `ShallowWrapper {}`;

exports[`should render a title text 1`] = `ShallowWrapper {}`;

exports[`should render an search by filter 1`] = `ShallowWrapper {}`;

exports[`should render an search form 1`] = `ShallowWrapper {}`;

exports[`should render card container with genres 1`] = `ShallowWrapper {}`;

exports[`should render card container without genres 1`] = `ShallowWrapper {}`;

exports[`should render headerFooter component 1`] = `ShallowWrapper {}`;

exports[`should render movie card 1`] = `ShallowWrapper {}`;
